adagrams/game.py

```diff
@@ -1,11 +1,139 @@
+    #appears as value
+    dict_word = word_letter_dictionaries(word_upper)
+    dict_letter_bank = word_letter_dictionaries(letter_bank)
+    #for loop to loop through dict_word  and check whether the letters are in dict_letter_bank
+    #or not and in case the amount of that letter exceeds the amount in dict_letter_bank it returns False
+    for elem in dict_word:
+        if elem not in letter_bank or dict_word[elem] > dict_letter_bank[elem]:
+            valid = False
+            break
+        else:
+            valid = True
+
+    return valid
 
 def score_word(word):
-    pass
+    #create dict_score to store letters that shares the same point value
+    dict_score = {1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+                  2: ["D", "G"],
+                  3: ["B", "C", "M", "P"],
+                  4: ["F", "H", "V", "W", "Y"],
+                  5: ["K"],
+                  8: ["J", "X"],
+                  10: ["Q", "Z"]}
+    #score variable to store the points earned
+    result = 0
+    #word_upper_case variable converts word variable elements into uppercase letters
+    word_upper = word.upper()
+    #for to loop through the elements in word_upper to give the right points per letter
+    for letter in word_upper:
+        for score, letters in dict_score.items():
+            if letter in letters:
+                result += score
+            
+    #if word has 7, 8, 9 or 10 letters the word gets 8 additional points
+    if len(word) >= 7 and len(word)<= 10:
+        result += 8
+        
+    return result
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    #tuple to store the word with the high score
+    high_score_tuple = ()
+    #dictionary to store all the words with their score
+    score_dict = {}
+    score = 0
+    #append every word with its score in the dictionary
+    for word in word_list:
+        score = score_word(word)
+        score_dict[word]= score
+    #get the word with the maximun score in score_dict
+    value_high_score = max(score_dict.values())
+    #append the words that share the same store into list_high_word
+    list_high_word = [word for word, score in score_dict.items() if score == value_high_score]    
+    #if there are only one word with the max score no need to loop through the list
+    if len(list_high_word) == 1:
+        high_score_tuple = (list_high_word[0], value_high_score)
+    #loop through list_high_word to get the word and value that should be stored in high_score_tuple
+    for word in range(len(list_high_word)):
+        last_word = list_high_word[word-1]
+        if len(list_high_word[word]) == 10:
+            high_score_tuple = (list_high_word[word], value_high_score)
+            break
+        elif len(list_high_word[word]) < len(last_word):
+            high_score_tuple = (list_high_word[word], value_high_score)
```

Comment:
👀 We don't want to check that the current word is better (shorter) than the word that _immediately_ preceeded it. We want to check whether it's better than the _best so far_.

Otherwise, these checks look great.